pkg/disruption/backend/sampler/remote.go

```diff
@@ -0,0 +1,501 @@
+package sampler
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	projv1 "github.com/openshift/api/project/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	projectclient "github.com/openshift/client-go/project/clientset/versioned"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	monitorserialization "github.com/openshift/origin/pkg/monitor/serialization"
+)
+
+const (
+	namespace            = "e2e-disruption-monitor"
+	serviceAccountName   = "disruption-monitor-sa"
+	varLogPath           = "/var/log"
+	disruptionDataFolder = "disruption-data"
+	disruptionTypeEnvVar = "DISRUPTION_TYPE_LABEL"
+	inClusterEventsFile  = "junit/AdditionalEvents__in_cluster_disruption.json"
+)
+
+var disruptionDataPath = fmt.Sprintf("%s/%s", varLogPath, disruptionDataFolder)
+
+func TearDownInClusterMonitors(config *rest.Config) error {
```

Comment:
rename.  teardown should be about deleting the namespace containing the pods you started.

Reply:
Done. Hopefully it would stop daemonsets before destroying serviceaccount (previously it was recorded as disruption event)